core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/support/IncludeExclude.java

```diff
@@ -138,6 +169,22 @@ public boolean accept(BytesRef value) {
 
     }
 
+    class PartitionedOrdinalsFilter extends OrdinalsFilter {
+        @Override
+        public LongBitSet acceptedGlobalOrdinals(RandomAccessOrds globalOrdinals) throws IOException {
+            final long numOrds=globalOrdinals.getValueCount();
+
+            LongBitSet acceptedGlobalOrdinals = new LongBitSet(numOrds);
+            final int zeroBasedPartition = incPartition-1;
+            for (int i = 0; i < numOrds; i++) {
+                if (i % incNumPartitions == zeroBasedPartition){                
+                    acceptedGlobalOrdinals.set(i);
+                }
+            }
```

Comment:
+1 for the terms enum pulled from the sorted set. It should be fast to enumerate the terms since it is a sequential reads in the term dict of each segment. The hashing will be slow though, but again it's great if we can have a way to exhaust a term aggregations consistently even if it's not as fast as we would like ;).